ksql-engine/src/main/java/io/confluent/ksql/analyzer/ExpressionAnalyzer.java

```diff
@@ -138,5 +122,32 @@ public Object visitQualifiedNameReference(
       throwOnUnknownField(node.getName());
       return null;
     }
+
+    private void throwOnUnknownField(final QualifiedName name) {
+      final Set<String> sourcesWithField = sourceSchemas.sourcesWithField(name.name());
+      if (sourcesWithField.isEmpty()) {
+        if (allowWindowMetaFields && name.name().equals(SchemaUtil.WINDOWSTART_NAME)) {
```

Comment:
Why do you check only for window start?

Reply:
Because that's the only windows meta field at the moment.